taglib/riff/wav/wavproperties.h

```diff
@@ -52,35 +52,107 @@ namespace TagLib {
         /*!
          * Create an instance of WAV::Properties with the data read from the
          * ByteVector \a data.
+         *
+         * \deprecated
          */
         Properties(const ByteVector &data, ReadStyle style);
 
         /*!
          * Create an instance of WAV::Properties with the data read from the
          * ByteVector \a data and the length calculated using \a streamLength.
+         *
+         * \deprecated
          */
         Properties(const ByteVector &data, uint streamLength, ReadStyle style);
 
         /*!
+         * Create an instance of WAV::Properties with the data read from the
+         * ByteVector \a data and the length calculated using \a streamLength
+         * and \a totalSamples.
+         *
+         * \note totalSamples can be zero if the file doesn't have a 'fact' chunk.
+         */
+        Properties(const ByteVector &data, uint streamLength, uint totalSamples, ReadStyle style);
```

Comment:
Hmm.  I'm not really excited about adding more constructors with [magic parameters](http://doc.qt.digia.com/qq/qq13-apis.html#theconveniencetrap) here.  I realize I'm responsible for a lot of them in the code, but that's mostly in code that I wrote more than 10 years ago.  In new code (and in TagLib 2.x) I'd prefer to move to something more along the lines of:

    Properties &setTotalSamples(uint samples);

Since there you can chain these things like:

    Properties p = Properties(data).setStreamLength(length).setTotalSamples(samples);